python/nav/ipdevpoll/plugins/entity.py

```diff
@@ -70,7 +70,7 @@ def _need_to_collect(self):
     def _process_entities(self, result):
         """Process the list of collected entities."""
         # be able to look up all entities using entPhysicalIndex
-        entities = EntityTable(result)
+        entities = dict(sorted(EntityTable(result).items()))
         containers = [
             self._container_from_entity(entity) for entity in entities.values()
         ]
```

Comment:
While this does the trick in this particular case, I'm not a big fan of discarding the `EntityTable` object with a pure `dict`.

To me, the EntityTable doesn't need to be sorted itself, while the following intuits that we just need to process its contents in a sorted order:

```suggestion
        entities = EntityTable(result)
        containers = [
            self._container_from_entity(entity) for _, entity in sorted(entities.items())
        ]
```

Reply:
There's also the more "involved" suggestion of:
```suggestion
        entities = EntityTable(result)
        containers = [
            self._container_from_entity(entity) for entity in sorted(entities.values(), key=operator.itemgetter(0))
        ]
```

which requires an extra import. There are three places where the row index is available: 
1. The EntityTable dictionary key
2. The entity value's `0` key
3. The entity value's `entPhysicalIndex` key.

But I like being able to sort by 1 without needing to involve the others :)

Reply:
Also, the fact that dict contents are sorted by insertion order is mostly an implementation detail of CPython, and it hasn't always been so - which is why my gut reflex is never to rely on the built-in ordering of dict objects.